lib/reissue/fragment_reader.rb

```diff
@@ -0,0 +1,43 @@
+require "pathname"
+
+module Reissue
+  class FragmentReader
+    VALID_SECTIONS = %w[added changed deprecated removed fixed security].freeze
+
+    def initialize(fragment_directory = "changelog.d")
+      @fragment_directory = Pathname.new(fragment_directory)
+    end
+
+    def read
+      return {} unless @fragment_directory.exist?
+
+      fragments = {}
+
+      @fragment_directory.glob("*.*.md").each do |fragment_file|
+        filename = fragment_file.basename.to_s
+        parts = filename.split(".")
+
+        next unless parts.length == 3
+
+        section = parts[1].downcase
+        next unless VALID_SECTIONS.include?(section)
```

Comment:
Prompt: FragmentReader should not own control over what are valid sections. Update the implementation to receive `valid_sections:` as an argument for the initializer. Instead, validity should be controlled at a higher level but injected as a dependency. Set the default value to nil and allow all sections if the value is nil or if it is an array, only allow sections which match the array.